hetzner_ddns.openwrt.rc

```diff
@@ -4,9 +4,24 @@
 START=99
 USE_PROCD=1
 
+_LOG_FILE=/var/log/hetzner_ddns.log
+_PID_FILE=/var/run/hetzner_ddns.pid
+
+fix_permissions() {
```

Comment:
Should we also set the owner of the config file to `nobody:nogroup`?

I did encounter issues when leaving it to `root:root`. 
What I did:
- changed the config file contents in order to match my zone on hetzner
- restarted the service with `service hetzner_ddns restart`
- had a look in my log file which showed me the following:
```
[2024-12-27 19:52:00] Started Hetzner DDNS daemon
[2024-12-27 19:52:00] Error: unable to read configuration file /etc/config/hetzner_ddns.conf
```
- after changing the owner of `hetzner_ddns.conf` to `nobody:nogroup` I could process the config file, because the log showed me this:
```
[2024-12-27 20:26:07] Started Hetzner DDNS daemon
[2024-12-27 20:26:07] Reading configuration from /etc/config/hetzner_ddns.conf
[2024-12-27 20:26:07] Info: update of A records not set, enabling by default
[2024-12-27 20:26:07] Info: update of AAAA records not set, enabling by default
[2024-12-27 20:26:09] Zone for refractedlight.eu: <redacted>
[2024-12-27 20:26:09] IPv4 record for cloud.refractedlight.eu: <redacted>
[2024-12-27 20:26:09] IPv6 record for cloud.refractedlight.eu: (missing)
[2024-12-27 20:26:09] Configuration successful
[2024-12-27 20:26:09] Watching for IP address and record changes
```
- then I also could confirm that the process is running:
```
 # ps | grep hetzner
17873 nobody    1564 S    {hetzner_ddns} /bin/sh /usr/bin/hetzner_ddns -d
```

But maybe this was just something strange on my installation... I am not that sure. 
Did test that with 
[hetzner-ddns_0.2.7-1_all.ipk.txt](https://github.com/user-attachments/files/18263052/hetzner-ddns_0.2.7-1_all.ipk.txt)
 (`.txt` suffix only because of GitHub rejecting pure `.ipk`)

Reply:
This is strange. I will give it some further investigations in the course of the weekend, but what permissions did your config file have after installation/editing?

Reply:
Sorry for the fuss, seems that I did something wrong previously or had some older configuration files.
<details><summary>Installation history</summary>
<code>
root@danielcpe ~
 # opkg install hetzner-ddns_0.2.7-1_all.ipk 
Installing hetzner-ddns (0.2.7-1) to root...
Configuring hetzner-ddns.
</code>

<code>
root@danielcpe /etc/config
 # ll hetzner_ddns.conf
-rw-r--r--    1 root     root           453 Dec 27 14:34 hetzner_ddns.conf
</code>
<code>
root@danielcpe /etc/config
 # cat hetzner_ddns.conf
interval=60                             # Seconds between updates / TTL value
key='********************************'  # Hetzner DNS API key
domain='example.com'                    # Top level domain name
records='homelab media vpn'             # Space separated host subdomains (@ for domain itself)
ipv4=true                               # Enable updating A records (IPv4)
ipv6=true                               # Enable updating AAAA records (IPv6)
</code>
<code>
root@danielcpe /etc/config
 # cat /var/log/hetzner_ddns.log 
[2024-12-29 17:14:55] Started Hetzner DDNS daemon
[2024-12-29 17:14:55] Reading configuration from /etc/config/hetzner_ddns.conf
[2024-12-29 17:14:56] Error: Invalid API key
</code>
</details> 

Now the configuration file is owned by root, but the daemon process is running and owned by nobody and the update works: 
```
16841 nobody    1564 S    {hetzner_ddns} /bin/sh /usr/bin/hetzner_ddns -d
```
```
[2024-12-29 17:32:34] Watching for IP address and record changes
[2024-12-29 17:36:14] Update IPv4 for cloud.refractedlight.eu: 127.0.0.1 => 84.135.98.106
```

TL;DR Seems fine now :heavy_check_mark: